### PR TITLE
`max-train-steps` support for `YOLO` integration

### PR DESCRIPTION
**Pull request by rahul-tuli**
The goal of this PR is to add `max_train_steps` support to our yolov5 integration. 

This will allow shorter test runs for validation

### CODE REVIEW

File: src/sparseml/pytorch/object_detection/train.py
Comment (bfineran) [inline src/sparseml/pytorch/object_detection/train.py:881]: nit: underscore names for args not supported in this script

File: src/sparseml/pytorch/object_detection/train.py
Comment (dbogunowicz) [inline src/sparseml/pytorch/object_detection/train.py:465]: Could you give some context to what happens here?

**Review by dbogunowicz — APPROVED** [review]

File: src/sparseml/pytorch/object_detection/train.py
Comment (bfineran) [inline src/sparseml/pytorch/object_detection/train.py:465]: what happens if max_train_steps is not set

**Review by bfineran — CHANGES_REQUESTED** [review]

File: src/sparseml/pytorch/object_detection/train.py
Comment (rahul-tuli) [reply src/sparseml/pytorch/object_detection/train.py:465]: Sure!

So `enumerate(iterable)` in python returns an iterator over an iterable, along with the index of the element, 

The goal of this PR is to only perform training for a fixed number of batches a.k.a `max_train_steps`, an easy way to accomplish this is to limit the "batch_iterator" to yield only first `max_train_steps` number of batches.

Line 465 accomplishes just that, It creates a new generator object, which would raise a `StopIteraton` exception after `max_train_steps` this allows us to use the newly created generator directly in for loops. Does that make sense?

**Review by bfineran — CHANGES_REQUESTED** [review]
also need to add support for max_eval_steps

**Review by bfineran — DISMISSED** [review]
LGTM, must wait for https://github.com/neuralmagic/yolov5/pull/41 to land and be built into the released wheel before landing

**Review by bfineran — APPROVED** [review]

**Comment by bfineran** [comment]
update looks good. reiterating - "must wait for https://github.com/neuralmagic/yolov5/pull/41 to land and be built into the released wheel before landing"

**Review by KSGulin — DISMISSED** [review]
LGTM!

**Review by bfineran — DISMISSED** [review]

**Comment by rahul-tuli** [comment]
Noting: Do not land until latest yolov5 wheel has been pushed

**Review by dbogunowicz — DISMISSED** [review]

**Review by bfineran — APPROVED** [review]

File: src/sparseml/pytorch/object_detection/train.py
Comment (dbogunowicz) [inline src/sparseml/pytorch/object_detection/train.py:584]:
```suggestion
                LOGGER.info(f"\nStopping early, {train_steps_executed} training steps executed")
```

**Review by dbogunowicz — APPROVED** [review]

**Review by dbogunowicz — APPROVED** [review]

**Review by dbogunowicz — APPROVED** [review]

**Review by KSGulin — APPROVED** [review]

**Review by bfineran — APPROVED** [review]

**Pull request merged**
